src/app/core/auth/auth.module.ts

```diff
@@ -0,0 +1,39 @@
+import { APP_INITIALIZER, InjectionToken, NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { OAuthModule, OAuthService, OAuthStorage } from 'angular-oauth2-oidc';
+import { AuthService } from './auth.service';
+import { environment } from 'src/environments/environment';
+import { EnvironmentType } from 'src/environments/model';
+import { MemoryService } from './memory.service';
+import { AuthGuard } from './auth.guard';
+
+function authFactory(authService: AuthService): () => Promise<boolean> {
+  return () =>
+    [EnvironmentType.OFFAUTH].includes(environment.environmentType)
+      ? Promise.resolve(true)
+      : authService
+          .initAuth()
+          .catch((_) => Promise.resolve(true))
+          .then((_) => Promise.resolve(true));
```

Comment:
It doesn't matter a lot here as the code is simple but usually it's better to put the `catch` block after `then` so that any error happening in the `then` callback is also caught.

Reply:
done